median/plugin.go

```diff
@@ -46,7 +46,8 @@ func (p *Plugin) NewMedianFactory(ctx context.Context, provider types.MedianProv
 		GasPriceSubunitsDataSource:           gasPriceSubunits,
 		IncludeGasPriceSubunitsInObservation: includeGasPriceSubunitsInObservation,
 		Logger: logger.NewOCRWrapper(lggr, true, func(msg string) {
-			ctx, cancelFn := p.stop.NewCtx()
+			var cancelFn context.CancelFunc
+			ctx, cancelFn = p.stop.NewCtx()
```

Comment:
We need to use a different name here instead of reassigning the `ctx` from the surrounding scope.

Reply:
idk what to do here, this same ctx seems to be used below, but it always throws shadowed declaration

Reply:
if I use a diff name it can't be used outside of this scope

Reply:
although it doesn't seem like this wrapped ctx should live outside of this scope